cpp/src/umap/simpl_set_embed/optimize_batch_kernel.cuh

```diff
@@ -325,7 +541,7 @@ __global__ void optimize_batch_kernel(
   }
   // storing gradients for positive samples back to global memory
   if (use_shared_mem) {
-    __syncthreads();
+    cub::CTA_SYNC();
     if (multicore_implem) {
```

Comment:
Since each vertex is now able to update its gradients independently, we can also merge the code paths here. We should also be able to apply the gradients directly to the output and no longer need the 2-step approach using the separate array to collect the outputs.